ui/components/app/snaps/snap-permission-cell/snap-permission-cell.js

```diff
@@ -0,0 +1,132 @@
+import PropTypes from 'prop-types';
+import { SubjectType } from '@metamask/permission-controller';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  AvatarBase,
+  AvatarBaseSize,
+  AvatarFavicon,
+  AvatarFaviconSize,
+  IconSize,
+  Text,
+} from '../../../component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  Display,
+  FontWeight,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import {
+  getAvatarFallbackLetter,
+  getSnapName,
+} from '../../../../helpers/utils/util';
+import PermissionCell from '../../permission-cell';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { getSnapsMetadata } from '../../../../selectors';
+
+export default function SnapPermissionCell({
+  snapId,
+  showOptions,
+  permission,
+  connectionSubjectMetadata,
+  index,
+  revoked,
+  approved,
+}) {
+  const t = useI18nContext();
+
+  let { label, description, leftIcon } = permission;
+
+  const snapsMetadata = useSelector(getSnapsMetadata);
+
+  if (permission.connection) {
+    if (connectionSubjectMetadata?.subjectType === SubjectType.Snap) {
+      const snapName = getSnapName(snapsMetadata)(
```

Comment:
Again cleaner to extract some stuff into functions.

Reply:
Refactored / updated.